give.php

```diff
@@ -386,6 +401,12 @@ public function init() {
 			$this->comment                = Give_Comment::get_instance();
 			$this->session_db             = new Give_DB_Sessions();
 			$this->session                = Give_Session::get_instance();
+			$this->themes                 = new Themes();
```

Comment:
@ravinderk I think this is fine for now as it's the convention, but I want to point one thing out.

The major difference between this and a dependency container is that the container acts in a JIT (Just In Time) fashion, instantiating classes _as they're needed_. What this is doing is instantiating everything up front. So whether or not a class instance is actually used it's instantiated for _every_ request. This can potentially create a good deal of unnecessary overhead. Make sense?

Reply:
@JasonTheAdams Yes. Do you have any idea to handle it other than this way

Reply:
As I said, since this is the the way it works now, I'm fine with this. But in the future we should really make a [container](https://laravel.com/docs/6.x/container).